src/btreemap/node.rs

```diff
@@ -474,14 +458,47 @@ impl NodeHeader {
     }
 }
 
-// The value in a K/V pair.
+/// The value in a K/V pair.
 #[derive(Debug)]
-enum Value {
-    // The value's encoded bytes.
-    ByVal(Vec<u8>),
+pub(crate) struct Value {
```

Comment:
Just to confirm: do we have enough test coverage to catch cases where the invariant we list would be violated?

Reply:
The constructors or `Value` ensured that this was already the case, however, I moved back to an enum in order to encode it in the type system.

Reply:
Even better to enforce it on the type system level, thanks!